rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -587,6 +599,50 @@ rmw_create_publisher(
     } \
   }
 
+#define SET_SUBMESSAGE_VALUE(dynamic_data, i) \
+  DDS_DynamicData sub_dynamic_data(NULL, DDS_DYNAMIC_DATA_PROPERTY_DEFAULT); \
+  DDS_ReturnCode_t status = dynamic_data->bind_complex_member( \
+    sub_dynamic_data, \
+    NULL, \
+    i + 1); \
+  if (status != DDS_RETCODE_OK) { \
+    rmw_set_error_string("failed to bind complex member"); \
+    return false; \
+  } \
+  void * sub_ros_message = (void *)((char *)ros_message + member->offset_); \
+  if (!member->members_) { \
+    rmw_set_error_string("members handle is null"); \
+    return false; \
+  } \
+  const::rosidl_typesupport_introspection_cpp::MessageMembers * sub_members = \
+    (const::rosidl_typesupport_introspection_cpp::MessageMembers *)member->members_->data; \
```

Comment:
That would be better. Since the code block was only moved in the PR from below into a macro I didn't change it. I will address the C casts in another round.